commands/creds.py

```diff
@@ -0,0 +1,108 @@
+import click
+import html
+import os
+import sys
+import taskcluster
+import urllib.parse
+import urllib.request
+import webbrowser
+import wsgiref.simple_server
+
+
+DEFAULT_ROOT_URL = "https://community-tc.services.mozilla.com/"
+DEFAULT_SCOPES = ["assume:project-admin:servo"]
+
+
+@click.command()
+@click.option("-p", "--port", default=0,
+              help="Port number to listen to. Zero selects on at random.")
+@click.option("-n", "--name", default="cli",
+              help="An identifier to include in the client ID")
+@click.option("-d", "--description", default="Temporary client for use on the command line",
+              help="The new client’s description")
+@click.option("-e", "--expires", default="2d",
+              help="How much time until the credentials expire")
+@click.option("-s", "--scopes", default=DEFAULT_SCOPES, multiple=True,
+              help="Which Taskcluster scopes to grant to the new client")
+def creds(port, name, description, expires, scopes):
+    """
+    Create a new Taskcluster “client” for command-line use,
+    and print environment variables with its credentials.
+
+    Use it like this:
+
+    $ eval `./mach creds`
+    """
+    root_url = os.environ.get("TASKCLUSTER_ROOT_URL", DEFAULT_ROOT_URL)
+    client_id = None
+    access_token = None
+    missing_scopes = None
+    def wsgi_app(environ, start_response):
+        nonlocal client_id
+        nonlocal access_token
+        nonlocal missing_scopes
+        query = urllib.parse.parse_qs(environ['QUERY_STRING'])
+        client_id, = query["clientId"]
+        access_token, = query["accessToken"]
+
+        auth = taskcluster.Auth({
+            "rootUrl": root_url,
+            "credentials": {"clientId": client_id, "accessToken": access_token},
+        })
+        missing_scopes = set(scopes) - set(auth.currentScopes()["scopes"])
```

Comment:
I think this creates a new local variable for `wsgi_app`, rather than setting the `missing_scopes` from line 39 to a new value.  Does the check on line 100 work?

Reply:
Assigning to the variable from line 39 is what `nonlocal` does in Python 3

Reply:
Ah, I totally forgot about that feature!  I largely stopped doing Python development before Py3 was a common thing..